content/blog/teach-tidyverse-2021/index.Rmd

```diff
@@ -0,0 +1,437 @@
+
+-   <img src="lifecycle-stable.svg" alt="Stable" style="vertical-align:middle"/> Stable indicates that breaking changes will be avoided where possible, and they're only made if the long term benefit of such a change exceeds the short term pain of changing existing code.
+    If breaking changes are needed, they will occur gradually.
+    Note that this badge is generally not shown as it applies to a huge majority of functions.
+    Teaching tip: teach away any stable functions, they're here to stay for the long run!
+
+-   <img src="lifecycle-deprecated.svg" alt="Deprecated" style="vertical-align:middle"/> If a function is noted as deprecated, this means a better alternative is available and this function is scheduled for removal.
+    Generally functions will first be soft deprecated and then deprecated.
+    Very important functions that become deprecated might next be defunct, which means that function continues to exist but the deprecation warning turns into an error.
+    An example of a deprecated function is `tibble::data_frame()`, with the preferred alternative `tibble::tibble()`. Arguments to functions can also be deprecated, e.g., in `tidyr::nest()` the new argument `new_col` makes the former `.key` argument not needed, and hence `.key` is deprecated.
+    You should avoid teaching functions that are deprecated and correct their usage in your students' code by suggesting the preferred alternative.
+
+-   <img src="lifecycle-superseded.svg" alt="Superseded" style="vertical-align:middle"/> Superseded indicates that there is a known better alternative for the function, but it's not going away.
+    Some examples include the following:
+
+    -   `tidyr::spread()` / `tidyr::gather()` vs. `tidyr::pivot_longer()` / `tidyr::pivot_wider()`
+
+    -   Scoped verbs (e.g., `dplyr::mutate_if()`, `dplyr::select_at()`, `dplyr::rename_all()`, etc.) vs. `dplyr::across()`
+
+    -   `dplyr::sample_n()` / `dplyr::sample_frac()` vs. `dplyr::slice_sample()` with `n` and `prop` arguments
+
+    I would recommend not teaching superseded functions to new learners, and for learners who might be aware of them already, I would recommend discouraging their use (though not correcting, i.e., no point deductions on a formative assessment), and suggesting an alternative.
+
+-   <img src="lifecycle-experimental.svg" alt="Experimental" style="vertical-align:middle"/> Experimental functions are made available so the community can try them out and provide feedback, however they come with no promises for long term stability.
+    Some examples include the following:
+
+    -   `dplyr::group_trim()` to drop unused levels of all factors that are used as grouping variables
+
+    -   in `dplyr::summarize()`: `.groups` argument to define the grouping structure of the result
+
+    -   in `dplyr::mutate()`: `.before` and `.after` arguments to control where new columns should appear
```

Comment:
FWIW I'm pretty confident that the experimental badge will be removed from both of these in the next dplyr release. Maybe you could mention that in an aside somehow?